packages/core/e2e/Promotion.spec.ts

```diff
@@ -0,0 +1,30 @@
+import { AddCouponPreset, PaymentStepAsGuestPreset, test } from '@bigcommerce/checkout/test-framework';
+
+test.describe('Shipping with coupon', () => {
+    test('`Adding coupon` after shipping step is submitted shows shipping error modal', async ({ assertions, checkout, page }) => {
```

Comment:
Adding a coupon after shipping step does not always result in an error modal showing up.
Perhaps we should mention it is a free-shipping coupon.